yt/data_objects/static_output.py

```diff
@@ -339,6 +342,18 @@ def force_periodicity(self, val=True):
             raise TypeError("force_periodicity expected a boolean.")
         self._force_periodicity = val
 
+    def set_periodicity(self, val):
+        """
+        Set box periodicity to user defined periodicity.
+        """
+        if isinstance(val, bool):
+            val = (val, val, val)
+        if not isinstance(val, tuple):
+            raise TypeError("set_periodicity must be a boolean or tuple of booleans.")
+        if len(val) != 3 or not all([isinstance(v, bool) for v in val]):
```

Comment:
`all` accepts generator, no need for a list comprehension (albeit really cheap here)
```suggestion
        if len(val) != 3 or not all(isinstance(v, bool) for v in val):
```

Reply:
Whoops, good catch.